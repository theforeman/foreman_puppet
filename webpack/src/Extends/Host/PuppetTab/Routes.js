import PropTypes from 'prop-types';
import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import { route } from './helpers';
import Reports from './SubTabs/Reports';
import ENCPreview from './SubTabs/ENCPreview';

const SecondaryTabRoutes = ({ hostName, hostInfo, status }) => (
  <Switch ouiaId="foreman-puppet-switch">
    <Route path={route('reports')}>
      <Reports hostName={hostName} hostInfo={hostInfo} status={status} />
    </Route>
    {/* <Route path={route('assigned')}>
      <EmptyPage
        header={__('Assigned classes')}
        description={__('This tab is still a work in progress')}
      />
    </Route>
    <Route path={route('smart-classes')}>
      <EmptyPage
        header={__('Smart class parameters')}
        description={__('This tab is still a work in progress')}
      />
    </Route> */}
    <Route path={route('yaml')}>
      <ENCPreview hostName={hostName} />
    </Route>
    <Redirect to={route('reports')} />
  </Switch>
);

SecondaryTabRoutes.propTypes = {
  hostName: PropTypes.string,
  hostInfo: PropTypes.object,
  status: PropTypes.string,
};

SecondaryTabRoutes.defaultProps = {
  hostName: '',
  hostInfo: {},
  status: undefined,
};

export default SecondaryTabRoutes;
