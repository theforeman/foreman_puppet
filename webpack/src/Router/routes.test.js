import React from 'react';
import { shallow } from '@theforeman/test';

import Routes from './routes';

describe('ForemanPuppetRoutes', () => {
  it('should create routes', () => {
    Object.entries(Routes).forEach(([key, Route]) => {
      const Component = Route.component;
      const component = shallow(<Component history={{}} some="props" />);
      Route.renderResult = component;
    });

    expect(Routes).toMatchSnapshot();
  });
});
