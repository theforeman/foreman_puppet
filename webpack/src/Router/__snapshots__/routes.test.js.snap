// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForemanPuppetRoutes should create routes 1`] = `Object {}`;
